### ISSUE TITLE
Incorrect syntax highlighting in js

### ISSUE THREAD

**Issue by SzHeJason** [Microsoft/vscode#57942]
```js
X({
    getA() {
        let str = 'str' // After inserting newline,it will get correct color
        if(str){
            console.log("correct color")
        }
    },

    getB() {
        return Promise.all() 
    }
})
```
![image](https://user-images.githubusercontent.com/20609396/45077262-44bade80-b11f-11e8-862c-0bf47eea4525.png)

![image](https://user-images.githubusercontent.com/20609396/45077229-3076e180-b11f-11e8-82d0-c6a16c37da2e.png)

- VSCode Version: 1.26.1
- OS Version: MacOS 10.13.6

**Comment by beastdestroyer** [comment]
I believe this is how VSCode treats some of JavaScript's automatic semi-colon insertion, in your first example, VSCode seems to treat the `if` statement as part of the declaration of `let str` (even though I'm pretty sure JS Engines would correct that) and newlines essentially trigger an invisible semi-colon, which would validate that JS as in your second example.

Tip: Always put semi-colons after variables, it's hard to rely on automatic semi colon insertion, it can lead to nasty bugs!

**Comment by SzHeJason** [comment]
In JavaScript , it's popular  rule that semi is inserted at end of syntax

**Comment by beastdestroyer** [comment]
It is popular for both methods to be used, however both can cause errors if used incorrectly. In this case VSCode's JS Grammar is highlighting this incorrectly.

**Comment by mjbvz** [comment]
Seems fixed in VS Code insiders builds (and in 1.27 which will be released shortly):

<img width="749" alt="screen shot 2018-09-05 at 7 33 52 am" src="https://user-images.githubusercontent.com/12821956/45100308-123cc180-b0de-11e8-8906-92a9a8e70d6f.png">
